### PR TITLE
prefer-method-signature: Convert to function

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Overview of change:

Just a refactor.

### CODE REVIEW

File: src/rules/preferMethodSignatureRule.ts [hunk 1]
```diff
@@ -36,25 +37,26 @@ export class Rule extends Lint.Rules.AbstractRule {
...
-    private createMethodSignatureFix(node: ts.PropertyDeclaration, type: ts.FunctionTypeNode): Lint.Fix | undefined {
-        return type.type && [
-            this.deleteFromTo(Lint.childOfKind(node, ts.SyntaxKind.ColonToken)!.getStart(), type.getStart()),
-            this.deleteFromTo(Lint.childOfKind(type, ts.SyntaxKind.EqualsGreaterThanToken)!.getStart(), type.type.getStart()),
-            this.appendText(Lint.childOfKind(type, ts.SyntaxKind.CloseParenToken)!.end, ":"),
-        ];
-    }
+function createMethodSignatureFix(node: ts.PropertySignature, type: ts.FunctionTypeNode): Lint.Fix | undefined {
+    return type.type && [
+        Lint.Replacement.deleteFromTo(Lint.childOfKind(node, ts.SyntaxKind.ColonToken)!.getStart(), type.getStart()),
```
Comment (ajafff) [inline src/rules/preferMethodSignatureRule.ts]:
```ts
deleteFromTo(node.name.end, type.getStart())
```

File: src/rules/preferMethodSignatureRule.ts [hunk 2]
```diff
@@ -36,25 +37,26 @@ export class Rule extends Lint.Rules.AbstractRule {
...
+        Lint.Replacement.deleteFromTo(Lint.childOfKind(type, ts.SyntaxKind.EqualsGreaterThanToken)!.getStart(), type.type.getStart()),
```
Comment (ajafff) [inline src/rules/preferMethodSignatureRule.ts]:
```ts
replaceFromTo(Lint.childOfKind(type, ts.SyntaxKind.EqualsGreaterThanToken)!.pos, type.type.pos, ":")
```

File: src/rules/preferMethodSignatureRule.ts [hunk 3]
```diff
@@ -36,25 +37,26 @@ export class Rule extends Lint.Rules.AbstractRule {
...
+        Lint.Replacement.appendText(Lint.childOfKind(type, ts.SyntaxKind.CloseParenToken)!.end, ":"),
```
Comment (ajafff) [inline src/rules/preferMethodSignatureRule.ts]: is unnecessary if you take the suggestions from above

File: src/rules/preferMethodSignatureRule.ts (hunk 1 above)
Comment (andy-ms) [inline src/rules/preferMethodSignatureRule.ts]: Need this because there might be a `?`.

**Pull request merged**